test/lib/pwa.js

```diff
@@ -39,23 +39,39 @@ const MANIFEST_URL = 'https://www.terra.com.br/manifest-br.json';
 const START_URL = 'https://www.terra.com.br/?utm_source=homescreen';
 const MANIFEST_DATA = './test/manifests/icon-url-with-parameter.json';
 const LIGHTHOUSE_JSON_EXAMPLE = './test/lib/lighthouse-example.json';
+const MANIFEST_DATA_NO_ICON = './test/manifests/no-icon.json';
+const MANIFEST_DATA_INVALID = './test/manifests/invalid.json';
 
 describe('lib.pwa', () => {
   let manifest;
   let pwa;
   let lighthouse;
+  let pwaNoIcon;
+  let manifestInvalid;
+  let pwaInvalid;
   before(done => {
-    dataFetcher.readFile(MANIFEST_DATA)
+    Promise.all(
+      [dataFetcher.readFile(MANIFEST_DATA)
       .then(jsonString => {
         manifest = new Manifest(MANIFEST_URL, JSON.parse(jsonString));
         pwa = new Pwa(MANIFEST_URL, manifest);
         pwa.id = '123456789';
         dataFetcher.readFile(LIGHTHOUSE_JSON_EXAMPLE)
           .then(data => {
             lighthouse = new Lighthouse('123456789', 'www.domain.com', JSON.parse(data));
-            done();
           });
-      });
+      })],
+      [dataFetcher.readFile(MANIFEST_DATA_NO_ICON)
```

Comment:
I think so, I had a hard time escaping manifest properly to embed them in the test. I will try again.